src/base.jl

```diff
@@ -91,7 +91,7 @@ function getindex(K::AbstractKroneckerProduct, i1::Integer, i2::Integer)
     A, B = getmatrices(K)
     m, n = size(A)
     k, l = size(B)
-    return A[cld(i1, k), cld(i2, l)] * B[(i1 - 1) % k + 1, (i2 - 1) % l + 1]
+    return (A[cld(i1, k), cld(i2, l)]::eltype(A)) * (B[(i1 - 1) % k + 1, (i2 - 1) % l + 1]::eltype(B))
```

Comment:
type-assertions here help if the individual `getindex` fails to infer the type, eg. in `KroneckerPower`